floodlight/io/datasets.py

```diff
@@ -799,3 +801,278 @@ def _download_matches_info(self) -> None:
+    division. A detailed description of the dataset as well as the collection process
+    can be found in the accompanying paper. Data for one match can be queried calling
+    the :func:`~DFLDataset.get`-method by specifying the match. The following matches
+    are available::
+
+        matches = {
+        'J03WMX': 1. FC Köln vs. FC Bayern München,
+        'J03WN1': VfL Bochum 1848 vs. Bayer 04 Leverkusen,
+        'J03WPY': Fortuna Düsseldorf vs. 1. FC Nürnberg,
+        'J03WOH': Fortuna Düsseldorf vs. SSV Jahn Regensburg,
+        'J03WQQ': Fortuna Düsseldorf vs. FC St. Pauli,
+        'J03WOY': Fortuna Düsseldorf vs. F.C. Hansa Rostock,
+        'J03WR9': Fortuna Düsseldorf vs. 1. FC Kaiserslautern
+        }
+
+    Examples
+    --------
+    >>> from floodlight.io.datasets import IDSSEDataset
+
+    >>> dataset = IDSSEDataset()
+    # get one sample
+    >>> events_data_objects, position_data_objects = dataset.get()
+    # get the corresponding pitch
+    >>> pitch = dataset.get_pitch()
+
+
+    References
+    ----------
+        .. [1] `Bassek, M., Müller-Budack, E., Ewerth, R., Weber, H., Rein, R., &
+            Memmert,D. (2024). An integrated dataset of synchronized spatiotemporal and
+            event data in elite soccer. In Submission.
+    """
+
+    def __init__(self, dataset_dir_name="idsse_dataset", match_id="J03WMX"):
+        self._IDSSE_SCHEMA = "https"
+        self._IDSSE_BASE_URL = "figshare.com/ndownloader/files"
+        self._IDSSE_FILE_IDS_INFO = {
+            "J03WMX": "48392485",
+            "J03WN1": "48392491",
+            "J03WPY": "48392497",
+            "J03WOH": "48392515",
+            "J03WQQ": "48392488",
+            "J03WOY": "48392503",
+            "J03WR9": "48392494",
+        }
+        self._IDSSE_FILE_IDS_EVENT = {
+            "J03WMX": "48392524",
+            "J03WN1": "48392527",
+            "J03WPY": "48392542",
+            "J03WOH": "48392500",
+            "J03WQQ": "48392521",
+            "J03WOY": "48392518",
+            "J03WR9": "48392530",
+        }
+        self._IDSSE_FILE_IDS_POSITION = {
+            "J03WMX": "48392539",
+            "J03WN1": "48392512",
+            "J03WPY": "48392572",
+            "J03WOH": "48392578",
+            "J03WQQ": "48392545",
+            "J03WOY": "48392551",
+            "J03WR9": "48392563",
+        }
+        self._IDSSE_PRIVAT_LINK = "1f806cb3e755c6b54e05"
+        if match_id in self._IDSSE_FILE_IDS_INFO.keys():
+            self._IDSSE_HOST_URL_INFO = f"{self._IDSSE_SCHEMA}://{self._IDSSE_BASE_URL}/{self._IDSSE_FILE_IDS_INFO[match_id]}?private_link={self._IDSSE_PRIVAT_LINK}"
+            self._IDSSE_HOST_URL_EVENT = f"{self._IDSSE_SCHEMA}://{self._IDSSE_BASE_URL}/{self._IDSSE_FILE_IDS_EVENT[match_id]}?private_link={self._IDSSE_PRIVAT_LINK}"
+            self._IDSSE_HOST_URL_POSITION = f"{self._IDSSE_SCHEMA}://{self._IDSSE_BASE_URL}/{self._IDSSE_FILE_IDS_POSITION[match_id]}?private_link={self._IDSSE_PRIVAT_LINK}"
```

Comment:
how does black fix max line length here?

Reply:
It simply doesn't. This file was already formatted by black. I added some line breaks to tidy it up. Note, that black formatted some stuff in the `StatsbombOpenDataset` in this commit.